.github/workflows/publish.yaml

```diff
@@ -103,13 +107,49 @@ jobs:
       - name: Run tests
         run: pytest
 
+  push:
+    name: Push client
+    needs:
+      - test
+    runs-on: ubuntu-latest
+
+    steps:
+      # Step 1: Checkout the repository
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Step 2: Download the package
+      - name: Download the package
+        uses: actions/download-artifact@v3
+        with:
+          name: python-package
+          path: ./
+
+      # Step 3: Get version from setup
+      - name: Get version from setup.py
+        id: get_version
+        run: |
+          version=$(python setup.py --version)
+          echo "version=$version" >> $GITHUB_ENV
```

Comment:
Can you add little comment it (maybe backed with a reference link) explaining what you're doing?